### PR TITLE
adding english-monarchs-and-marriages (enmoma) dataset

### PR DESCRIPTION
**Pull request by frankiethull**
creating the english-monarchs-and-marriages branch, aka enmoma for short. testing the pull request method with tidytuesday team on this old [issue 480](https://github.com/rfordatascience/tidytuesday/issues/480). 

one thing to note, I did not use ttsave()!, I'm not sure if that function exists or if I'm missing a step, please review `saving.R` script.

### CODE REVIEW

File: data/curated/enmoma/saving.R [hunk 1]
```diff
@@ -0,0 +1,9 @@
+source("data/curated/enmoma/cleaning.R")
```
Comment (jonthegeek) [inline data/curated/enmoma/saving.R:1]: When you run this step, it should add the `ttsave` function to your environment. Please try to use this and note any errors. The save step also creates the data dictionary file, and does some checks.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
Please try to run the `source()` line in `saving.R`, and let me know if that doesn't work. It SHOULD give you the `ttsave()` function, which (in addition to saving the dataset) creates the data dictionary and runs some (minimal for now but more to come) checks.

We clearly need to make this instruction clearer in any case!

File: data/curated/enmoma/english_monarchs_marriages_df.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+|variable         |class     |description      |
+|:----------------|:---------|:----------------|
+|king_name        |character |king_name        |
```
Comment (jonthegeek) [inline data/curated/enmoma/english_monarchs_marriages_df.md:3]: Edit the third column of each of these rows to add a description of the field.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
I can't give this a full review right now, so I'm sending this single comment. If you don't have a chance to update things right away, no worries. I'll try to dig into the rest of this submission sometime tomorrow.

Thank you so much for testing this process! You're the first person to learn the process purely from the instructions, and there are clearly some things for us to clarify!

File: data/curated/enmoma/english_monarchs_marriages_df.md (hunk 2 above)
Comment (lgibson7) [reply data/curated/enmoma/english_monarchs_marriages_df.md]: Yay! Thank you so much for your help @frankiethull.

Comment (frankiethull) [reply data/curated/enmoma/english_monarchs_marriages_df.md]: @lgibson7 thanks for the opportunity to contribute! I think this one is completed, glad to be the first person to try it out! All the steps are documented well I think I just have issues following directions :)

Comment (jonthegeek) [reply data/curated/enmoma/english_monarchs_marriages_df.md]: I'm at posit conf, but I'll try to take a close look tonight.

I think our instructions might be a bit too wordy right now. We need to make it harder to misread 😊

File: data/curated/enmoma/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+title: English Monarchs and Marriages | Names, Ages, and Historical Years
```
Comment (jonthegeek) [inline data/curated/enmoma/meta.yaml:1]:
```suggestion
title: English Monarchs and Marriages
```
The extra-long title might trip up social media posts, and I'll need to specifically watch for (and/or better deal with) `|` characters! That would break the tables of datasets.

File: data/curated/enmoma/intro.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# English Monarchs and Marriages 
+
+This week we are exploring [English Monarchs and Marriages](https://github.com/frankiethull/english_monarch_marriages)! 
+
+> this dataset focuses on the names, ages, and marriages of various 'kings' and 'consorts'. the data ranges all the way back to 850 where the details are a bit fuzzy, spanning all the way to current day. names contain special characters; ages & years can be a bit of a regex puzzle as well. additionally, the age of kings and consorts may show quite a bit of an age gap. 
+
+The data was scraped from [Ian Visits](https://www.ianvisits.co.uk/articles/a-list-of-monarchs-by-marriage-6857/).
```
Comment (jonthegeek) [inline data/curated/enmoma/intro.md:7]:
```suggestion
The data was scraped from [Ian Visits](https://www.ianvisits.co.uk/articles/a-list-of-monarchs-by-marriage-6857/) by [f. hull](https://github.com/frankiethull), who also curated this week's post!
```

Comment (jonthegeek) [reply data/curated/enmoma/intro.md:7]: See #725 for a ticket to make crediting more-automatic. Does this credit look ok, @frankiethull?

File: data/curated/enmoma/meta.yaml (hunk 3 above)
Comment (jonthegeek) [reply data/curated/enmoma/meta.yaml:1]: Addressed in #726 (checks to deal with special characters) and #727 (improve instructions about title).

**Comment by lgibson7** [comment]
Yay! Our first (correct me if I'm wrong @jonthegeek) externally curated Tidy Tuesday data set. Thank you so much @frankiethull. I truly appreciate your help with this.

**Comment by jonthegeek** [comment]
> Yay! Our first (correct me if I'm wrong @jonthegeek) externally curated Tidy Tuesday data set. Thank you so much @frankiethull. I truly appreciate your help with this. 

Not first ever, but first using the new, formal system. The one or two we've had in the past ended up needing a lot of manual edits to fit the format, since we didn't have any guidance on what to do. So this is still a very "yay!" moment!

**Comment by frankiethull** [comment]
"yay!" moment confirmed over here!! Glad to be your guinea pig and that credit is perfectly fine with me. 

I use the vertical bar "|" often so this is a helpful note for next time. 

I had shared a handful of datasets back in 2022, which looks like they will all be closed issues now (woohoo!). But with this new process, I'll be on the lookout for other interesting datasets. I also saw you all did a rebranding as DSLC, congratulations, I have been out of the loop! 

Additionally saw this, "We hope to provide instructions for other programming languages soon", I could always be a tester for Julia and Python too, let me know when and where :triumph:

File: data/2024/2024-08-20/readme.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# English Monarchs and Marriages
+
```
Comment (jonthegeek) [inline data/2024/2024-08-20/readme.md:2]:
```suggestion
```

Comment (jonthegeek) [reply data/2024/2024-08-20/readme.md:2]: Oops, I forgot  that the script add this. Another thing to make more clear in instructions!

**Review by jonthegeek — APPROVED** [review]
I have to commit one fix that I didn't catch (but it became obvious after the script to assign this to a week). Thanks again! Your contribution will appear as next week's dataset!
